src/DDTrace/Integrations/CakePHP/CakePHPIntegration.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace DDTrace\Integrations\CakePHP;
+
+use DDTrace\Configuration;
+use DDTrace\Integrations\CakePHP\V2\CakePHPIntegrationLoader;
+use DDTrace\Integrations\Integration;
+
+/**
+ * The base Laravel integration which delegates loading to the appropriate integration version.
+ */
+class CakePHPIntegration extends Integration
+{
+    const NAME = 'cakephp';
+
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * Loads the integration.
+     *
+     * @return int
+     */
+    public static function load()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return self::NOT_AVAILABLE;
+        }
+
+        $integration = self::getInstance();
+
+        // CakePHP v2.x - we don't need to check for v3 since it does not have \Dispatcher or \ShellDispatcher
+        $initCakeV2 = function () use ($integration) {
+            $loader = new CakePHPIntegrationLoader();
+            $loader->load($integration);
+            return dd_trace_forward_call();
+        };
+        // Web bootstrap
+        dd_trace('Dispatcher', '__construct', $initCakeV2);
+        // CLI bootstrap
+        //dd_trace('ShellDispatcher', 'run', $initCakeV2);
+        // Temporary workaround until we fix request_init_hook for non-autoloaded projects
```

Comment:
This temporary might be quite permanent :) 
I would drop that comment for now, but of course, we should discuss alternatives :D

Reply:
Yeah, that comment isn't super clear. I'd like to keep something there as a note to change it once we get auto-instrumentation rocking for non-autoloaded projects. So I've updated the comment to be a bit more clear. Does that work for you? :)

Reply:
What I meant more is that - I've seen a lot of comments in the code where `temporary` was still there 4+ years later. 

In itself saying something is temporary without having a good plan and timeline of getting rid of this temporary is just asking for a case like this. 

Its more a philosophical issue than engineering one. though :)

Reply:
Very good point! 😄